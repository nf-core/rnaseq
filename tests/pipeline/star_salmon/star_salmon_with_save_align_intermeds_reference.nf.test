nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    options "-profile test,docker"
    tag "pipeline"

    test("STAR Salmon (Save Alignment Intermediate BAM & Reference) (GSE110004)") {

        when {
            params {
                outdir               = "$outputDir"
                aligner              = "star_salmon"
                save_align_intermeds = true
                save_reference       = true
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(path("$outputDir/pipeline_info/software_versions.yml")).match("software_versions") },
                { assert snapshot(path("$outputDir/pipeline_info/samplesheet.valid.csv")).match("samplesheet") },
                { assert new File("$outputDir/bbsplit/RAP1_IAA_30M_REP1.stats.txt").exists() },
                { assert new File("$outputDir/bbsplit/RAP1_UNINDUCED_REP1.stats.txt").exists() },
                { assert new File("$outputDir/bbsplit/RAP1_UNINDUCED_REP2.stats.txt").exists() },
                { assert new File("$outputDir/bbsplit/WT_REP1.stats.txt").exists() },
                { assert new File("$outputDir/bbsplit/WT_REP2.stats.txt").exists() },
                { assert snapshot(path("$outputDir/genome/genes.gtf"),
                                path("$outputDir/genome/genome_gfp.bed"),
                                path("$outputDir/genome/genome_gfp.fasta"),
                                path("$outputDir/genome/genome_gfp.fasta.fai"),
                                path("$outputDir/genome/genome_gfp.fasta.sizes"),
                                path("$outputDir/genome/genome_gfp.gtf"),
                                path("$outputDir/genome/gfp.fa")).match("genome") },
                { assert new File("$outputDir/fastqc/RAP1_IAA_30M_REP1_1_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/RAP1_IAA_30M_REP1_2_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/RAP1_UNINDUCED_REP1_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/RAP1_UNINDUCED_REP2_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/WT_REP1_1_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/WT_REP1_2_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/WT_REP2_1_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/WT_REP2_2_fastqc.html").exists() },
                { assert snapshot(path("$outputDir/multiqc/star_salmon/multiqc_data/junction_saturation_known.txt"),
                                path("$outputDir/multiqc/star_salmon/multiqc_data/junction_saturation_novel.txt"),
                                path("$outputDir/multiqc/star_salmon/multiqc_data/multiqc_fail_strand_check_table.txt"),
                                path("$outputDir/multiqc/star_salmon/multiqc_data/multiqc_fastqc.txt")).match("multiqc") },
                { assert new File("$outputDir/multiqc/star_salmon/multiqc_data/multiqc_featurecounts_biotype_plot.txt").exists() },
                { assert new File("$outputDir/multiqc/star_salmon/multiqc_data/multiqc_dupradar-plot.txt").exists() },
                { assert new File("$outputDir/multiqc/star_salmon/multiqc_data/multiqc_general_stats.txt").exists() },
                { assert new File("$outputDir/multiqc/star_salmon/multiqc_data/multiqc_salmon.txt").exists() },
                { assert new File("$outputDir/multiqc/star_salmon/multiqc_data/multiqc_star.txt").exists() },
                { assert new File("$outputDir/multiqc/star_salmon/multiqc_data/rseqc_inner_distance.txt").exists() },
                { assert new File("$outputDir/multiqc/star_salmon/multiqc_data/rseqc_read_dups.txt").exists() },
                { assert new File("$outputDir/multiqc/star_salmon/multiqc_data/rseqc_junction_saturation_all.txt").exists() },
                { assert new File("$outputDir/multiqc/star_salmon/multiqc_data/multiqc_rseqc_infer_experiment.txt").exists() },
                { assert new File("$outputDir/multiqc/star_salmon/multiqc_data/multiqc_rseqc_bam_stat.txt").exists() },
                { assert new File("$outputDir/multiqc/star_salmon/multiqc_data/multiqc_rseqc_junction_annotation.txt").exists() },
                { assert new File("$outputDir/multiqc/star_salmon/multiqc_data/multiqc_rseqc_read_distribution.txt").exists() },
                { assert new File("$outputDir/multiqc/star_salmon/multiqc_data/multiqc_samtools_flagstat.txt").exists() },
                { assert new File("$outputDir/multiqc/star_salmon/multiqc_data/multiqc_samtools_idxstats.txt").exists() },
                { assert new File("$outputDir/multiqc/star_salmon/multiqc_data/multiqc_samtools_stats.txt").exists() },
                { assert new File("$outputDir/multiqc/star_salmon/multiqc_data/qualimap_rnaseq_cov_hist.txt").exists() },
                { assert new File("$outputDir/multiqc/star_salmon/multiqc_data/qualimap_rnaseq_genome_results.txt").exists() },
                { assert snapshot(path("$outputDir/salmon/salmon_tx2gene.tsv")).match("salmon") },
                { assert new File("$outputDir/salmon/RAP1_IAA_30M_REP1/quant.sf").exists() },
                { assert new File("$outputDir/salmon/RAP1_IAA_30M_REP1/quant.genes.sf").exists() },
                { assert new File("$outputDir/salmon/RAP1_UNINDUCED_REP1/quant.sf").exists() },
                { assert new File("$outputDir/salmon/RAP1_UNINDUCED_REP1/quant.genes.sf").exists() },
                { assert new File("$outputDir/salmon/RAP1_UNINDUCED_REP2/quant.sf").exists() },
                { assert new File("$outputDir/salmon/RAP1_UNINDUCED_REP2/quant.genes.sf").exists() },
                { assert new File("$outputDir/salmon/WT_REP1/quant.sf").exists() },
                { assert new File("$outputDir/salmon/WT_REP1/quant.genes.sf").exists() },
                { assert new File("$outputDir/salmon/WT_REP2/quant.sf").exists() },
                { assert new File("$outputDir/salmon/WT_REP2/quant.genes.sf").exists() },
                { assert new File("$outputDir/salmon/salmon.merged.gene_counts_length_scaled.rds").exists() },
                { assert new File("$outputDir/salmon/salmon.merged.gene_counts.rds").exists() },
                { assert new File("$outputDir/salmon/salmon.merged.gene_counts_scaled.rds").exists() },
                { assert new File("$outputDir/salmon/salmon.merged.transcript_counts.rds").exists() },
                { assert snapshot(path("$outputDir/star_salmon/salmon_tx2gene.tsv")).match("star_salmon") },
                { assert new File("$outputDir/star_salmon/RAP1_IAA_30M_REP1.Aligned.out.bam").exists() },
                { assert new File("$outputDir/star_salmon/RAP1_IAA_30M_REP1.Aligned.toTranscriptome.out.bam").exists() },
                { assert new File("$outputDir/star_salmon/RAP1_IAA_30M_REP1.sorted.bam").exists() },
                { assert new File("$outputDir/star_salmon/RAP1_IAA_30M_REP1.sorted.bam.bai").exists() },
                { assert new File("$outputDir/star_salmon/RAP1_IAA_30M_REP1.markdup.sorted.bam").exists() },
                { assert new File("$outputDir/star_salmon/RAP1_IAA_30M_REP1.markdup.sorted.bam.bai").exists() },
                { assert new File("$outputDir/star_salmon/WT_REP1.Aligned.out.bam").exists() },
                { assert new File("$outputDir/star_salmon/WT_REP1.Aligned.toTranscriptome.out.bam").exists() },
                { assert new File("$outputDir/star_salmon/WT_REP1.sorted.bam").exists() },
                { assert new File("$outputDir/star_salmon/WT_REP1.sorted.bam.bai").exists() },
                { assert new File("$outputDir/star_salmon/WT_REP1.markdup.sorted.bam").exists() },
                { assert new File("$outputDir/star_salmon/WT_REP1.markdup.sorted.bam.bai").exists() },
                { assert new File("$outputDir/star_salmon/RAP1_UNINDUCED_REP1.Aligned.out.bam").exists() },
                { assert new File("$outputDir/star_salmon/RAP1_UNINDUCED_REP1.Aligned.toTranscriptome.out.bam").exists() },
                { assert new File("$outputDir/star_salmon/RAP1_UNINDUCED_REP1.sorted.bam").exists() },
                { assert new File("$outputDir/star_salmon/RAP1_UNINDUCED_REP1.sorted.bam.bai").exists() },
                { assert new File("$outputDir/star_salmon/RAP1_UNINDUCED_REP1.markdup.sorted.bam").exists() },
                { assert new File("$outputDir/star_salmon/RAP1_UNINDUCED_REP1.markdup.sorted.bam.bai").exists() },
                { assert new File("$outputDir/star_salmon/WT_REP2.Aligned.out.bam").exists() },
                { assert new File("$outputDir/star_salmon/WT_REP2.Aligned.toTranscriptome.out.bam").exists() },
                { assert new File("$outputDir/star_salmon/WT_REP2.sorted.bam").exists() },
                { assert new File("$outputDir/star_salmon/WT_REP2.sorted.bam.bai").exists() },
                { assert new File("$outputDir/star_salmon/WT_REP2.markdup.sorted.bam").exists() },
                { assert new File("$outputDir/star_salmon/WT_REP2.markdup.sorted.bam.bai").exists() },
                { assert new File("$outputDir/star_salmon/RAP1_UNINDUCED_REP2.Aligned.out.bam").exists() },
                { assert new File("$outputDir/star_salmon/RAP1_UNINDUCED_REP2.Aligned.toTranscriptome.out.bam").exists() },
                { assert new File("$outputDir/star_salmon/RAP1_UNINDUCED_REP2.sorted.bam").exists() },
                { assert new File("$outputDir/star_salmon/RAP1_UNINDUCED_REP2.sorted.bam.bai").exists() },
                { assert new File("$outputDir/star_salmon/RAP1_UNINDUCED_REP2.markdup.sorted.bam").exists() },
                { assert new File("$outputDir/star_salmon/RAP1_UNINDUCED_REP2.markdup.sorted.bam.bai").exists() },
                { assert new File("$outputDir/star_salmon/salmon.merged.gene_counts_length_scaled.tsv").exists() },
                { assert new File("$outputDir/star_salmon/salmon.merged.gene_tpm.tsv").exists() },
                { assert new File("$outputDir/star_salmon/salmon.merged.gene_counts_scaled.tsv").exists() },
                { assert new File("$outputDir/star_salmon/salmon.merged.transcript_counts.tsv").exists() },
                { assert new File("$outputDir/star_salmon/salmon.merged.gene_counts.tsv").exists() },
                { assert new File("$outputDir/star_salmon/salmon.merged.transcript_tpm.tsv").exists() },
                { assert new File("$outputDir/star_salmon/RAP1_IAA_30M_REP1/quant.sf").exists() },
                { assert new File("$outputDir/star_salmon/RAP1_IAA_30M_REP1/quant.genes.sf").exists() },
                { assert new File("$outputDir/star_salmon/RAP1_UNINDUCED_REP1/quant.sf").exists() },
                { assert new File("$outputDir/star_salmon/RAP1_UNINDUCED_REP1/quant.genes.sf").exists() },
                { assert new File("$outputDir/star_salmon/RAP1_UNINDUCED_REP2/quant.sf").exists() },
                { assert new File("$outputDir/star_salmon/RAP1_UNINDUCED_REP2/quant.genes.sf").exists() },
                { assert new File("$outputDir/star_salmon/WT_REP1/quant.sf").exists() },
                { assert new File("$outputDir/star_salmon/WT_REP1/quant.genes.sf").exists() },
                { assert new File("$outputDir/star_salmon/WT_REP2/quant.sf").exists() },
                { assert new File("$outputDir/star_salmon/WT_REP2/quant.genes.sf").exists() },
                { assert new File("$outputDir/star_salmon/salmon.merged.gene_counts_length_scaled.rds").exists() },
                { assert new File("$outputDir/star_salmon/salmon.merged.gene_counts_scaled.rds").exists() },
                { assert new File("$outputDir/star_salmon/salmon.merged.gene_counts.rds").exists() },
                { assert new File("$outputDir/star_salmon/salmon.merged.transcript_counts.rds").exists() },
                { assert new File("$outputDir/star_salmon/bigwig/RAP1_IAA_30M_REP1.forward.bigWig").exists() },
                { assert new File("$outputDir/star_salmon/bigwig/RAP1_IAA_30M_REP1.reverse.bigWig").exists() },
                { assert new File("$outputDir/star_salmon/bigwig/RAP1_UNINDUCED_REP1.forward.bigWig").exists() },
                { assert new File("$outputDir/star_salmon/bigwig/RAP1_UNINDUCED_REP1.reverse.bigWig").exists() },
                { assert new File("$outputDir/star_salmon/bigwig/RAP1_UNINDUCED_REP2.forward.bigWig").exists() },
                { assert new File("$outputDir/star_salmon/bigwig/RAP1_UNINDUCED_REP2.reverse.bigWig").exists() },
                { assert new File("$outputDir/star_salmon/bigwig/WT_REP2.forward.bigWig").exists() },
                { assert new File("$outputDir/star_salmon/bigwig/WT_REP2.reverse.bigWig").exists() },
                { assert snapshot(path("$outputDir/star_salmon/deseq2_qc/R_sessionInfo.log")).match("star_salmon/deseq2_qc/") },
                { assert new File("$outputDir/star_salmon/deseq2_qc/deseq2.dds.RData").exists() },
                { assert new File("$outputDir/star_salmon/deseq2_qc/deseq2.pca.vals.txt").exists() },
                { assert new File("$outputDir/star_salmon/deseq2_qc/deseq2.plots.pdf").exists() },
                { assert new File("$outputDir/star_salmon/deseq2_qc/deseq2.sample.dists.txt").exists() },
                { assert new File("$outputDir/star_salmon/dupradar/gene_data/RAP1_IAA_30M_REP1_dupMatrix.txt").exists() },
                { assert new File("$outputDir/star_salmon/dupradar/gene_data/RAP1_UNINDUCED_REP1_dupMatrix.txt").exists() },
                { assert new File("$outputDir/star_salmon/dupradar/gene_data/RAP1_UNINDUCED_REP2_dupMatrix.txt").exists() },
                { assert new File("$outputDir/star_salmon/dupradar/gene_data/WT_REP1_dupMatrix.txt").exists() },
                { assert new File("$outputDir/star_salmon/dupradar/gene_data/WT_REP2_dupMatrix.txt").exists() },
                { assert new File("$outputDir/star_salmon/dupradar/intercepts_slope/RAP1_IAA_30M_REP1_intercept_slope.txt").exists() },
                { assert new File("$outputDir/star_salmon/dupradar/intercepts_slope/RAP1_UNINDUCED_REP1_intercept_slope.txt").exists() },
                { assert new File("$outputDir/star_salmon/dupradar/intercepts_slope/RAP1_UNINDUCED_REP2_intercept_slope.txt").exists() },
                { assert new File("$outputDir/star_salmon/dupradar/intercepts_slope/WT_REP1_intercept_slope.txt").exists() },
                { assert new File("$outputDir/star_salmon/dupradar/intercepts_slope/WT_REP2_intercept_slope.txt").exists() },
                { assert new File("$outputDir/star_salmon/featurecounts/RAP1_IAA_30M_REP1.featureCounts.txt").exists() },
                { assert new File("$outputDir/star_salmon/featurecounts/RAP1_UNINDUCED_REP1.featureCounts.txt").exists() },
                { assert new File("$outputDir/star_salmon/featurecounts/RAP1_UNINDUCED_REP2.featureCounts.txt").exists() },
                { assert new File("$outputDir/star_salmon/featurecounts/WT_REP1.featureCounts.txt").exists() },
                { assert new File("$outputDir/star_salmon/featurecounts/WT_REP2.featureCounts.txt").exists() },
                { assert new File("$outputDir/star_salmon/log/RAP1_IAA_30M_REP1.Log.final.out").exists() },
                { assert new File("$outputDir/star_salmon/log/RAP1_UNINDUCED_REP1.Log.final.out").exists() },
                { assert new File("$outputDir/star_salmon/log/RAP1_UNINDUCED_REP2.Log.final.out").exists() },
                { assert new File("$outputDir/star_salmon/log/WT_REP1.Log.final.out").exists() },
                { assert new File("$outputDir/star_salmon/log/WT_REP2.Log.final.out").exists() },
                { assert new File("$outputDir/star_salmon/picard_metrics/RAP1_IAA_30M_REP1.markdup.sorted.MarkDuplicates.metrics.txt").exists() },
                { assert new File("$outputDir/star_salmon/picard_metrics/RAP1_UNINDUCED_REP1.markdup.sorted.MarkDuplicates.metrics.txt").exists() },
                { assert new File("$outputDir/star_salmon/picard_metrics/RAP1_UNINDUCED_REP2.markdup.sorted.MarkDuplicates.metrics.txt").exists() },
                { assert new File("$outputDir/star_salmon/picard_metrics/WT_REP1.markdup.sorted.MarkDuplicates.metrics.txt").exists() },
                { assert new File("$outputDir/star_salmon/picard_metrics/WT_REP2.markdup.sorted.MarkDuplicates.metrics.txt").exists() },
                { assert new File("$outputDir/star_salmon/qualimap/RAP1_IAA_30M_REP1/rnaseq_qc_results.txt").exists() },
                { assert new File("$outputDir/star_salmon/qualimap/RAP1_UNINDUCED_REP1/rnaseq_qc_results.txt").exists() },
                { assert new File("$outputDir/star_salmon/qualimap/RAP1_UNINDUCED_REP2/rnaseq_qc_results.txt").exists() },
                { assert new File("$outputDir/star_salmon/qualimap/WT_REP1/rnaseq_qc_results.txt").exists() },
                { assert new File("$outputDir/star_salmon/qualimap/WT_REP2/rnaseq_qc_results.txt").exists() },
                { assert new File("$outputDir/star_salmon/rseqc/bam_stat/RAP1_IAA_30M_REP1.bam_stat.txt").exists() },
                { assert new File("$outputDir/star_salmon/rseqc/bam_stat/RAP1_UNINDUCED_REP1.bam_stat.txt").exists() },
                { assert new File("$outputDir/star_salmon/rseqc/bam_stat/RAP1_UNINDUCED_REP2.bam_stat.txt").exists() },
                { assert new File("$outputDir/star_salmon/rseqc/bam_stat/WT_REP1.bam_stat.txt").exists() },
                { assert new File("$outputDir/star_salmon/rseqc/bam_stat/WT_REP2.bam_stat.txt").exists() },
                { assert new File("$outputDir/star_salmon/rseqc/infer_experiment/RAP1_IAA_30M_REP1.infer_experiment.txt").exists() },
                { assert new File("$outputDir/star_salmon/rseqc/infer_experiment/RAP1_UNINDUCED_REP1.infer_experiment.txt").exists() },
                { assert new File("$outputDir/star_salmon/rseqc/infer_experiment/RAP1_UNINDUCED_REP2.infer_experiment.txt").exists() },
                { assert new File("$outputDir/star_salmon/rseqc/infer_experiment/WT_REP1.infer_experiment.txt").exists() },
                { assert new File("$outputDir/star_salmon/rseqc/infer_experiment/WT_REP2.infer_experiment.txt").exists() },
                { assert new File("$outputDir/star_salmon/rseqc/junction_annotation/bed/RAP1_IAA_30M_REP1.junction.bed").exists() },
                { assert new File("$outputDir/star_salmon/rseqc/junction_annotation/bed/RAP1_UNINDUCED_REP1.junction.bed").exists() },
                { assert new File("$outputDir/star_salmon/rseqc/junction_annotation/bed/RAP1_UNINDUCED_REP2.junction.bed").exists() },
                { assert new File("$outputDir/star_salmon/rseqc/junction_annotation/bed/WT_REP1.junction.bed").exists() },
                { assert new File("$outputDir/star_salmon/rseqc/junction_annotation/bed/WT_REP2.junction.bed").exists() },
                { assert new File("$outputDir/star_salmon/rseqc/junction_annotation/log/RAP1_IAA_30M_REP1.junction_annotation.log").exists() },
                { assert new File("$outputDir/star_salmon/rseqc/junction_annotation/log/WT_REP1.junction_annotation.log").exists() },
                { assert new File("$outputDir/star_salmon/rseqc/junction_annotation/log/WT_REP2.junction_annotation.log").exists() },
                { assert new File("$outputDir/star_salmon/rseqc/read_distribution/RAP1_IAA_30M_REP1.read_distribution.txt").exists() },
                { assert new File("$outputDir/star_salmon/rseqc/read_distribution/RAP1_UNINDUCED_REP1.read_distribution.txt").exists() },
                { assert new File("$outputDir/star_salmon/rseqc/read_distribution/RAP1_UNINDUCED_REP2.read_distribution.txt").exists() },
                { assert new File("$outputDir/star_salmon/rseqc/read_distribution/WT_REP1.read_distribution.txt").exists() },
                { assert new File("$outputDir/star_salmon/rseqc/read_distribution/WT_REP2.read_distribution.txt").exists() },
                { assert new File("$outputDir/star_salmon/rseqc/inner_distance/txt/RAP1_IAA_30M_REP1.inner_distance.txt").exists() },
                { assert new File("$outputDir/star_salmon/rseqc/inner_distance/txt/WT_REP1.inner_distance.txt").exists() },
                { assert new File("$outputDir/star_salmon/rseqc/inner_distance/txt/WT_REP2.inner_distance.txt").exists() },
                { assert new File("$outputDir/star_salmon/rseqc/junction_saturation/pdf/RAP1_IAA_30M_REP1.junctionSaturation_plot.pdf").exists() },
                { assert new File("$outputDir/star_salmon/rseqc/junction_saturation/pdf/RAP1_UNINDUCED_REP1.junctionSaturation_plot.pdf").exists() },
                { assert new File("$outputDir/star_salmon/rseqc/junction_saturation/pdf/RAP1_UNINDUCED_REP2.junctionSaturation_plot.pdf").exists() },
                { assert new File("$outputDir/star_salmon/rseqc/inner_distance/txt/RAP1_IAA_30M_REP1.inner_distance.txt").exists() },
                { assert new File("$outputDir/star_salmon/rseqc/inner_distance/txt/WT_REP1.inner_distance.txt").exists() },
                { assert new File("$outputDir/star_salmon/rseqc/inner_distance/txt/WT_REP2.inner_distance.txt").exists() },
                { assert new File("$outputDir/star_salmon/rseqc/junction_saturation/pdf/RAP1_IAA_30M_REP1.junctionSaturation_plot.pdf").exists() },
                { assert new File("$outputDir/star_salmon/rseqc/junction_saturation/pdf/RAP1_UNINDUCED_REP1.junctionSaturation_plot.pdf").exists() },
                { assert new File("$outputDir/star_salmon/rseqc/junction_saturation/pdf/RAP1_UNINDUCED_REP2.junctionSaturation_plot.pdf").exists() },
                { assert new File("$outputDir/star_salmon/samtools_stats/RAP1_IAA_30M_REP1.markdup.sorted.bam.flagstat").exists() },
                { assert new File("$outputDir/star_salmon/samtools_stats/RAP1_IAA_30M_REP1.markdup.sorted.bam.idxstats").exists() },
                { assert new File("$outputDir/star_salmon/samtools_stats/RAP1_IAA_30M_REP1.markdup.sorted.bam.stats").exists() },
                { assert new File("$outputDir/star_salmon/samtools_stats/RAP1_IAA_30M_REP1.sorted.bam.flagstat").exists() },
                { assert new File("$outputDir/star_salmon/samtools_stats/RAP1_IAA_30M_REP1.sorted.bam.idxstats").exists() },
                { assert new File("$outputDir/star_salmon/samtools_stats/RAP1_IAA_30M_REP1.sorted.bam.stats").exists() },
                { assert new File("$outputDir/star_salmon/samtools_stats/RAP1_UNINDUCED_REP1.markdup.sorted.bam.flagstat").exists() },
                { assert new File("$outputDir/star_salmon/samtools_stats/RAP1_UNINDUCED_REP1.markdup.sorted.bam.idxstats").exists() },
                { assert new File("$outputDir/star_salmon/samtools_stats/RAP1_UNINDUCED_REP1.markdup.sorted.bam.stats").exists() },
                { assert new File("$outputDir/star_salmon/samtools_stats/RAP1_UNINDUCED_REP1.sorted.bam.flagstat").exists() },
                { assert new File("$outputDir/star_salmon/samtools_stats/RAP1_UNINDUCED_REP1.sorted.bam.idxstats").exists() },
                { assert new File("$outputDir/star_salmon/samtools_stats/RAP1_UNINDUCED_REP1.sorted.bam.stats").exists() },
                { assert new File("$outputDir/star_salmon/samtools_stats/RAP1_UNINDUCED_REP2.markdup.sorted.bam.flagstat").exists() },
                { assert new File("$outputDir/star_salmon/samtools_stats/RAP1_UNINDUCED_REP2.markdup.sorted.bam.idxstats").exists() },
                { assert new File("$outputDir/star_salmon/samtools_stats/RAP1_UNINDUCED_REP2.markdup.sorted.bam.stats").exists() },
                { assert new File("$outputDir/star_salmon/samtools_stats/RAP1_UNINDUCED_REP2.sorted.bam.flagstat").exists() },
                { assert new File("$outputDir/star_salmon/samtools_stats/RAP1_UNINDUCED_REP2.sorted.bam.idxstats").exists() },
                { assert new File("$outputDir/star_salmon/samtools_stats/RAP1_UNINDUCED_REP2.sorted.bam.stats").exists() },
                { assert new File("$outputDir/star_salmon/samtools_stats/WT_REP1.markdup.sorted.bam.flagstat").exists() },
                { assert new File("$outputDir/star_salmon/samtools_stats/WT_REP1.markdup.sorted.bam.idxstats").exists() },
                { assert new File("$outputDir/star_salmon/samtools_stats/WT_REP1.markdup.sorted.bam.stats").exists() },
                { assert new File("$outputDir/star_salmon/samtools_stats/WT_REP1.sorted.bam.flagstat").exists() },
                { assert new File("$outputDir/star_salmon/samtools_stats/WT_REP1.sorted.bam.idxstats").exists() },
                { assert new File("$outputDir/star_salmon/samtools_stats/WT_REP1.sorted.bam.stats").exists() },
                { assert new File("$outputDir/star_salmon/samtools_stats/WT_REP2.markdup.sorted.bam.flagstat").exists() },
                { assert new File("$outputDir/star_salmon/samtools_stats/WT_REP2.markdup.sorted.bam.idxstats").exists() },
                { assert new File("$outputDir/star_salmon/samtools_stats/WT_REP2.markdup.sorted.bam.stats").exists() },
                { assert new File("$outputDir/star_salmon/samtools_stats/WT_REP2.sorted.bam.flagstat").exists() },
                { assert new File("$outputDir/star_salmon/samtools_stats/WT_REP2.sorted.bam.idxstats").exists() },
                { assert new File("$outputDir/star_salmon/samtools_stats/WT_REP2.sorted.bam.stats").exists() },
                { assert new File("$outputDir/star_salmon/stringtie/RAP1_IAA_30M_REP1.gene.abundance.txt").exists() },
                { assert new File("$outputDir/star_salmon/stringtie/RAP1_UNINDUCED_REP1.gene.abundance.txt").exists() },
                { assert new File("$outputDir/star_salmon/stringtie/RAP1_UNINDUCED_REP2.gene.abundance.txt").exists() },
                { assert new File("$outputDir/star_salmon/stringtie/WT_REP1.gene.abundance.txt").exists() },
                { assert new File("$outputDir/star_salmon/stringtie/WT_REP2.gene.abundance.txt").exists() },
                { assert new File("$outputDir/star_salmon/stringtie/RAP1_IAA_30M_REP1.coverage.gtf").exists() },
                { assert new File("$outputDir/star_salmon/stringtie/RAP1_UNINDUCED_REP1.coverage.gtf").exists() },
                { assert new File("$outputDir/star_salmon/stringtie/RAP1_UNINDUCED_REP2.coverage.gtf").exists() },
                { assert new File("$outputDir/star_salmon/stringtie/WT_REP1.coverage.gtf").exists() },
                { assert new File("$outputDir/star_salmon/stringtie/WT_REP2.coverage.gtf").exists() },
                { assert new File("$outputDir/star_salmon/stringtie/RAP1_IAA_30M_REP1.transcripts.gtf").exists() },
                { assert new File("$outputDir/star_salmon/stringtie/RAP1_UNINDUCED_REP1.transcripts.gtf").exists() },
                { assert new File("$outputDir/star_salmon/stringtie/RAP1_UNINDUCED_REP2.transcripts.gtf").exists() },
                { assert new File("$outputDir/star_salmon/stringtie/WT_REP1.transcripts.gtf").exists() },
                { assert new File("$outputDir/star_salmon/stringtie/WT_REP2.transcripts.gtf").exists() },
                { assert new File("$outputDir/trimgalore/RAP1_IAA_30M_REP1_1.fastq.gz_trimming_report.txt").exists() },
                { assert new File("$outputDir/trimgalore/RAP1_IAA_30M_REP1_2.fastq.gz_trimming_report.txt").exists() },
                { assert new File("$outputDir/trimgalore/RAP1_UNINDUCED_REP1.fastq.gz_trimming_report.txt").exists() },
                { assert new File("$outputDir/trimgalore/RAP1_UNINDUCED_REP2.fastq.gz_trimming_report.txt").exists() },
                { assert new File("$outputDir/trimgalore/WT_REP1_1.fastq.gz_trimming_report.txt").exists() },
                { assert new File("$outputDir/trimgalore/WT_REP1_2.fastq.gz_trimming_report.txt").exists() },
                { assert new File("$outputDir/trimgalore/WT_REP2_1.fastq.gz_trimming_report.txt").exists() },
                { assert new File("$outputDir/trimgalore/WT_REP2_2.fastq.gz_trimming_report.txt").exists() }
            )
        }

    }

}
