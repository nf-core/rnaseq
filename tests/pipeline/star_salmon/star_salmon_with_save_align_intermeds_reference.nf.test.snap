{
    "genome": {
        "content": [
            "genes.gtf:md5,f248c3bd0ed7d362da273417237da3b3",
            "genome_gfp.bed:md5,991993ebef1def1d5823632c21177ec3",
            "genome_gfp.fasta:md5,e23e302af63736a199985a169fdac055",
            "genome_gfp.fasta.fai:md5,8fa54c6bd2ea6a369efbb8ab4f30156a",
            "genome_gfp.fasta.sizes:md5,9b755f8f349b14accefb4d859f84de26",
            "genome_gfp.gtf:md5,9da10ce47264696420e2427f7a593ed9",
            "gfp.fa:md5,86d79ea5bd4a204d7108277dfd94824f"
        ],
        "timestamp": "2023-05-08T19:39:48+0000"
    },
    "samplesheet": {
        "content": [
            "samplesheet.valid.csv:md5,3d40bf3d424290cec0dbe9ee95b7dd12"
        ],
        "timestamp": "2023-05-08T19:39:48+0000"
    },
    "software_versions": {
        "content": [
            "software_versions.yml:md5,1dacbb94b6df6b5e05448e60967d828e"
        ],
        "timestamp": "2023-05-08T19:39:48+0000"
    },
    "star_salmon": {
        "content": [
            "salmon_tx2gene.tsv:md5,b367acfc4b7df7fb68fb1b1d2956e27e"
        ],
        "timestamp": "2023-05-08T19:39:48+0000"
    },
    "salmon": {
        "content": [
            "salmon_tx2gene.tsv:md5,b367acfc4b7df7fb68fb1b1d2956e27e"
        ],
        "timestamp": "2023-05-08T19:39:48+0000"
    },
    "multiqc": {
        "content": [
            "junction_saturation_known.txt:md5,7fb002deec1752d8e2065b693624191c",
            "junction_saturation_novel.txt:md5,7fb002deec1752d8e2065b693624191c",
            "multiqc_fail_strand_check_table.txt:md5,7797d69bfc52161c55e12f29e1bf51b4",
            "multiqc_fastqc.txt:md5,68bdc9b37911a79484e0333e82679fa3"
        ],
        "timestamp": "2023-05-08T19:39:48+0000"
    },
    "star_salmon/deseq2_qc/": {
        "content": [
            "R_sessionInfo.log:md5,fb0da0d7ad6994ed66a8e68348b19676"
        ],
        "timestamp": "2023-05-08T19:39:48+0000"
    }
}